### ISSUE TITLE
[Feature]: To make it easier to display the ellipsis when fluent-button's text overflows

### ISSUE THREAD

**Issue by HcySunYang** [microsoft/fluentui#31614]
### Library

Web Components (@fluentui/web-components)

Dependencies:

```
"@fluentui/tokens": "1.0.0-alpha.16",
"@fluentui/web-components": "3.0.0-beta.15",
"@microsoft/fast-element": "2.0.0-beta.26",
```

### Describe the feature that you would like added

For now, it is hard to display the ellipsis when the `fluent-button`'s text overflows. To achieve that, we have two approaches:

Approach 1: Live demo is here https://stackblitz.com/edit/typescript-bsh4pd?file=package.json,index.ts

```ts
import { css, customElement, html, FASTElement } from '@microsoft/fast-element';
import { DesignToken } from '@microsoft/fast-foundation';
import { setTheme } from '@fluentui/web-components';
import { webLightTheme } from '@fluentui/tokens';

import '@fluentui/web-components/button.js';

const template = html`
  <fluent-button appearance="primary">Change Your Password</fluent-button>
`;

const styles = css`
  fluent-button {
    width: 100px;
    min-width: 0;
  }
  fluent-button::part(content) {
    display: inline-block;
    width: 100%;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
  }
`;

@customElement({
  name: 'test-app',
  template,
  styles,
})
export class TestApp extends FASTElement {}

DesignToken.registerDefaultStyleTarget();

setTheme(webLightTheme);
```

With this approach, we have to give the fluent-button a `0` min-width since it is a flex box, and then change the `fluent-button::part(content)` from `flex` to `inline-block`. This approach is too invasive, it changes the origin box model of the internal elements of the fluent button, and it might not meet the needs of every case.

Approach 2: Live demo is here https://stackblitz.com/edit/typescript-1hpzfx?file=package.json,index.ts

```ts
import { css, customElement, html, FASTElement } from '@microsoft/fast-element';
import { DesignToken } from '@microsoft/fast-foundation';
import { setTheme } from '@fluentui/web-components';
import { webLightTheme } from '@fluentui/tokens';

import '@fluentui/web-components/button.js';

const template = html`
  <fluent-button><span class='text'>Change Your Password Please<span></fluent-button>
`;

const styles = css`
  fluent-button {
    width: 50px;
  }
  fluent-button::part(content) {
    min-width: 0;
  }
  .text {
    display: inline-block;
    width: 100%;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
  }
`;

@customElement({
  name: 'test-app',
  template,
  styles,
})
export class TestApp extends FASTElement {}

DesignToken.registerDefaultStyleTarget();

setTheme(webLightTheme);
```

With this approach, we introduced another element to wrap our text, it is `.text` in this case, and then we also need to set the `fluent-button::part(content)` a `0` min-width since it is a flex box. The downside of this approach is that we have to introduce another element.

### Validations

- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### Priority

Normal

**Comment by chrisdholt** [comment]
Thanks for this issue! I'd like to get @gouttierre's thoughts on this and/or perhaps get feedback from design. Currently, the design of the button is that text _wraps_. Supporting both wrapping and ellipsis seems counter-intuitive and difficult to me from a library standpoint. I don't know the exact design intent here, but I'd assume that truncating button text should ideally be a rare occurrence. Definitely understand the ask for convenience, first I'd like to get Gouttierre's thoughts :).

**Comment by gouttierre** [comment]
Hello; Apologies on the delay in getting back to this query. Our design team has reviewed the request. In short, this change would cause too much issues for the design system. 

- It puts undue burden on users who use zoom, since that will cause more buttons to truncate, or will truncate text even more, making it harder to guess what the buttons are for. Then those users will need to go through the UI and hover over every single item until they find the action they want (or figure out what their options are).
- People using pointers or pointer-like AT without hover (touch, voice control, eye control, switch setups, etc.) will be unable to access the tooltip, and just kinda out of luck.
- People using ZoomText and other zooming software (which is often paired with some form of static page zoom) have a harder time accessing tooltips, since they are using their mouse to move the visible portion of the screen. Tooltips mean they also have to keep their mouse in a specific position (i.e. over the button they want to read), which creates additional burden when trying to explore an interface.
- There's also the issue that tooltips cover other content, which is also annoying for zooming software users.

To summarize; our team will not be able to take this request. Thank you for reaching out.

cc @JustSlone , @tudorpopams , @chrisdholt, @paulgildea

**Comment by gouttierre** [comment]
@HcySunYang ~ Hello; Apologies on the delay in getting back to this query. Our design team has reviewed the request. In short, this change would cause too much issues for the design system.

- It puts undue burden on users who use zoom, since that will cause more buttons to truncate, or will truncate text even more, making it harder to guess what the buttons are for. Then those users will need to go through the UI and hover over every single item until they find the action they want (or figure out what their options are).
- People using pointers or pointer-like AT without hover (touch, voice control, eye control, switch setups, etc.) will be unable to access the tooltip, and just kind of out of luck.
- People using ZoomText and other zooming software (which is often paired with some form of static page zoom) have a harder time accessing tooltips, since they are using their mouse to move the visible portion of the screen. Tooltips mean they also have to keep their mouse in a specific position (i.e. over the button they want to read), which creates additional burden when trying to explore an interface.
- There's also the issue that tooltips cover other content, which is also annoying for zooming software users.

To summarize; our team will not be able to take this request. Thank you for reaching out.

cc @JustSlone , @tudorpopams , @chrisdholt, @paulgildea

**Comment by HcySunYang** [comment]
Sorry, I don't know what you are talking about, if I am the author of the fluent UI, I just need to add the following code, and the work is done:

```css
 fluent-button::part(content) {
    min-width: 0;
  }
```

The min-width supported by default by the fluent-button will make the work easier, it has nothing to do with the design you mentioned.
